dm/syncer/validate_worker.go

```diff
@@ -0,0 +1,380 @@
+// Copyright 2022 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package syncer
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"math"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/pingcap/errors"
+	"github.com/pingcap/tidb-tools/pkg/dbutil"
+	"github.com/pingcap/tidb/parser/model"
+	tidbmysql "github.com/pingcap/tidb/parser/mysql"
+	"go.uber.org/atomic"
+	"go.uber.org/zap"
+
+	"github.com/pingcap/tiflow/dm/dm/config"
+	tcontext "github.com/pingcap/tiflow/dm/pkg/context"
+	"github.com/pingcap/tiflow/dm/pkg/log"
+	"github.com/pingcap/tiflow/dm/pkg/terror"
+	"github.com/pingcap/tiflow/dm/syncer/dbconn"
+)
+
+const (
+	workerChannelSize = 1000
+
+	maxBatchSize = 100 // todo: choose a proper value, or configurable?
+)
+
+type validateFailedType int
+
+const (
+	hasDeletedRow validateFailedType = iota
+	rowNotExist
+	rowDifferent
+)
+
+type validateFailedRow struct {
+	tp      validateFailedType
+	dstData []*sql.NullString
+}
+
+type validateWorker struct {
+	cfg               config.ValidatorConfig
```

Comment:
Is it recommended to use pointer types here?

Reply:
both are ok, i think